pkg/configurator/crd_client.go

```diff
@@ -190,3 +250,83 @@ func meshConfigUpdatedMessageHandler(psubMsg *events.PubSubMessage) {
 			psubMsg.AnnouncementType)
 	}
 }
+
+func (c *CRDClient) getMeshConfigCacheKey() string {
+	return fmt.Sprintf("%s/%s", c.osmNamespace, c.meshConfigName)
+}
+
+// Returns the current ConfigMap
+func (c *CRDClient) getMeshConfig() *osmConfig {
```

Comment:
I don't see why we need the `osmConfig` struct anymore, given that `MeshConfig` is already a well defined type. The intermediary type conversion seems unnecessary and was only necessary because the ConfigMap only supported string key-value pairs.

Reply:
I agree. we can refactor this in a separate PR.